unlock-app/src/components/interface/user-account/SignUp.tsx

```diff
@@ -0,0 +1,233 @@
+import React from 'react'
+import styled from 'styled-components'
+import Link from 'next/link'
+import queryString from 'query-string'
+import { connect } from 'react-redux'
+import { signupEmail } from '../../../actions/user'
+import InvalidLink from '../InvalidLink'
+import SignupSuccess from '../SignupSuccess'
+import { Router, Account } from '../../../unlockTypes'
+import FinishSignUp from '../FinishSignup'
+import { verifyEmailSignature } from '../../../utils/wedlocks'
+
+interface Props {
+  signupEmail: (email: string) => any
+  toggleSignup: () => void
+  emailAddress?: string
+  isLinkValid?: boolean
+  account?: Account
+}
+
+interface State {
+  emailAddress: string
+  submitted: boolean // Used to handle intermediate state between submission and response
+}
+
+export class SignUp extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      emailAddress: '', // eslint-disable-line react/no-unused-state
+      submitted: false, // eslint-disable-line react/no-unused-state
+    }
+  }
+
+  handleSubmit = (event: any) => {
+    event.preventDefault()
+    const { signupEmail } = this.props
+    const { emailAddress } = this.state
+    signupEmail(emailAddress)
+    this.setState({
+      submitted: true, // eslint-disable-line react/no-unused-state
+    })
+  }
+
+  handleInputChange = (event: any) => {
+    this.setState({
+      emailAddress: event.target.value, // eslint-disable-line react/no-unused-state
+    })
+  }
+
+  handleClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    const { toggleSignup } = this.props
+    e.preventDefault()
+    toggleSignup()
+  }
+
+  render() {
+    const { submitted } = this.state
+    const { emailAddress, isLinkValid, account } = this.props
+
+    if (account && account.address) {
+      return <SignupSuccess />
+    }
+
+    const LogInLink = (callToAction: string) => (
+      <LinkButton onClick={this.handleClick}>{callToAction}</LinkButton>
+    )
+
+    if (!emailAddress) {
+      return (
+        <div>
```

Comment:
Not a blocker, but we should probably make future consideration to extract the return values here so that we can edit their make up without adjusting these files